ember-glimmer w/o jQuery initial work

wycats opened the pull request:
This commit gets the vast majority of tests in ember-glimmer working
without jQuery.

It creates a small test facility called NodeQuery that looks enough like
jQuery for most of the existing tests to work, but does not attempt to
use the facility outside of tests.

Once all tests pass, we should consider a quest issue to clean this up
further, but it's nice to get so many tests passing with such a huge
diff.

As in the previous pull request, you can run the tests without jQuery for a package by navigating to:

```
http://localhost:4200/tests/index.html?hidepassed&jquery=none&package=ember-glimmer
```

wycats committed:
ember-glimmer w/o jQuery initial work

This commit gets the vast majority of tests in ember-glimmer working
without jQuery.

It creates a small test facility called NodeQuery that looks enough like
jQuery for most of the existing tests to work, but does not attempt to
use the facility outside of tests.

Once all tests pass, we should consider a quest issue to clean this up
further, but it's nice to get so many tests passing with such a huge
diff.

wycats committed:
Clean up the code to not rely on newer features

rwjblue commented:
This is looking good.  I like the idea of the `NodeQuery` crutch, so that we can iterate on the cleanup efforts.

wycats committed:
Merge remote-tracking branch 'origin/master' into no-jquery

wycats committed:
Tests without nojquery all pass

As far as I can tell.

wycats committed:
Remove API no longer present in QUnit 2.0

wycats committed:
Merge remote-tracking branch 'origin/master' into no-jquery

wycats committed:
IE11 doesn't have Object.assign

Commit:
Avoid testing defaultPrevented on IE11.

Cross-reference from SO:
https://stackoverflow.com/questions/23349191/event-preventdefault-is-not-working-in-ie-11-for-custom-events

Basically it is not possible for us to test simulated events under I11.
This commit disables the only two related assertions when running under
IE11.

Commit:
Rename `noJqueryMode` to `jQueryDisabled`

rwjblue commented on packages/internal-test-helpers/lib/test-cases/abstract.js:
```diff
@@ -96,7 +98,19 @@ export default class AbstractTestCase {
 
   $(sel) {
-    return sel ? jQuery(sel, this.element) : jQuery(this.element);
+    if (sel instanceof Element) {
```
Should probably test for `nodeType`, good for future refactor...
